openedx/core/djangoapps/user_authn/views/tests/test_login.py

```diff
@@ -479,6 +484,45 @@ def test_check_password_policy_compliance_warning(self):
             self.assertIn('Test warning', self.client.session['_messages'])
         self.assertTrue(response_content.get('success'))
 
+    @ddt.data(
+        ('test_password', 'test_password', True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+    )
+    @ddt.unpack
+    @override_waffle_flag(PASSWORD_UNICODE_NORMALIZE, active=True)
+    def test_password_unicode_normalization_login(self, password, password_entered, login_success):
+        """
+        Tests unicode normalization on user's passwords on login.
+        """
+        self.user.set_password(password)
+        self.user.save()
+        response, mock_audit_log = self._login_response(self.user.email, password_entered,)
+        self._assert_response(response, success=login_success)
+
+    @ddt.data(
+        ('test_password', 'test_password', True),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKC', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), False),
+        (unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'),
+         unicodedata.normalize('NFKD', u'Ṗŕệṿïệẅ Ṯệẍt'), True),
+    )
+    @ddt.unpack
+    def test_password_unicode_normalization_login(self, password, password_entered, login_success):
```

Comment:
NIT: I'm not convinced this is a necessary test since it seems like something we will be throwing out with the removal of the waffle flag.

Reply:
Well... it's good to test that during rollout, we've coded the waffle flag correctly. Like what if we screwed up and turned on the feature for everyone regardless of waffle? This would catch that class of errors. Might only need one test case instead of 6? But still good to check, imho.

Reply:
Also, just noticed, these two tests have the same name so they aren't both being run. I believe it only runs the second defined function.